Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1494,6 +1503,22 @@ std::string DEntry::GCI_FileName() const
   return Common::EscapeFileName(filename);
 }
 
+void Header::FixChecksums()
+{
+  auto [checksum_sum, checksum_inv] = CalculateChecksums();
+  m_checksum = checksum_sum;
+  m_checksum_inv = checksum_inv;
+}
+
+std::pair<u16, u16> Header::CalculateChecksums() const
+{
+  static_assert(std::is_trivially_copyable<Header>::value);
+
+  std::array<u8, 0x1FC> raw;
```

Comment:
Thoughts on using a `constexpr` based on `Header` for this size instead? Something like `offsetof(Header, m_update_counter) - offsetof(Header, m_serial)` to make it clear what parts will be checksummed.
And if we decide to go with this, the same for the others.

Reply:
Yeah, that makes sense.

Reply:
Done.